src/redux/sagas/siteSaga.js

```diff
@@ -1,18 +1,43 @@
-import { put, takeLatest, takeEvery } from 'redux-saga/effects';
+import { takeEvery, put as dispatch } from 'redux-saga/effects';
+import { SITE_ACTIONS } from '../actions/siteActions';
+import { callSite, callSetSite } from '../requests/siteRequests';
 
 function* getSites() {
-
+  try {
+    const sites = yield callSite();
+      yield put({
+        type: SITE_ACTIONS.SET_SITES,
+        payload: sites,
+      });
+      yield put({
+        type: SITE_ACTIONS.FETCH_SITES,
+      });
+  } catch (error) {
+  console.log(`Error on getSites: ${error}`);
+  };
 }
 
 // set status of site
 // include action.payload with 'reset', 'spam', or 'notSpam'
-function* setSiteStatus() {
-
+function* setSiteStatus(payload) {
```

Comment:
Sagas take in actions, not payloads.

Reply:
Also, remember when you do this, you also have to update a reducer that holds the last three modified sites. It doesn't exist yet, but you should write that call in here.